AWS-Transcription.py

```diff
@@ -1,86 +1,102 @@
-from __future__ import print_function
+#!usr/bin/env python3
+
 import boto3
 import time
-import os
+from distutils.util import strtobool
+from pathlib import Path
 
 #Sets variables
 s3 = boto3.client('s3')
-print('Enter the s3 bucket name')
-s3Bucket = input()
-
-print('Enter name of file in Audio to upload')
-fileName = input()
-filePath = 'Audio/' + fileName
-noExt, fileExt = os.path.splitext(filePath)
-if fileExt == '.mp3':
-    fileExt = 'mp3'
-elif fileExt == '.mp4':
-    fileExt = 'mp4'
-elif fileExt == '.wav':
-    fileExt = 'wav'
-elif fileExt == '.flac':
-    fileExt = 'flac'
-else:
-    print('Invalid format. Convert to mp3 | mp4 | wav | flac')
-    exit()
+s3_bucket = input("Enter the s3 bucket name: ")
+
+file_name = input("Enter name of file in Audio to upload: ").strip()
+file_path = Path(__file__).parent / 'Audio' / file_name
+file_ext = file_path.suffix.lower()
+file_format = file_ext[1:]
+
+if file_format not in ('mp3', 'mp4', 'wav', 'flac'):
+    raise ValueError("Invalid format. Convert to mp3 | mp4 | wav | flac")
+    
+
+def bool_prompt(query):
+    """ Asks the user for input on a boolean question """
+    
+    while True: 
+        val = input(f"{query} [y/n]: ").lower()
+    
+        try:
+            ret = strtobool(val)
+            break
+        except ValueError:
+            print("Please answer with y/n")
+            
+    return ret
 
 #Uploads file from Audio
-print('Uploading')
-s3.upload_file(filePath,s3Bucket,fileName)
-
-#Checks that file has been uploaded - I believe that this section is uneccessary
-#but may be useful for longer upload times.
-while False:
-    s3.get_object(Bucket=s3Bucket,Key=fileName)
-    print('Still uploading')
-    time.sleep(5)
+print("Uploading")
+s3.upload_file(filePath,s3_bucket,fileName)
 
-print('Starting Transcription')
+print("Starting Transcription")
 
 ### Audio Transcription Start ###
 
-uri = 'https://s3-us-west-2.amazonaws.com/' + s3Bucket + '/' + fileName
```

Comment:
Changed the concatenated string to a formatted f-string.

Reply:
Thank you so much for all this input.  I literally just learned about f-string the other day.  I have learned a lot from this.  Thank again for helping all of us leaners.

Reply:
Just as a general rule, string formatting should usually be preferred over string concatenation in Python, unless the latter was somehow more readable (which can happen, but isn't common, and in such cases you need to be certain you're dealing with strings only).

Also, while f-strings are usually the best course of action for string literals, sometimes you need ready-made strings (in variables), in which case `str.format` should be used:

```python
foo = "spam"
bar = "eggs"
baz = "My favourite food is {} and {}"

print(baz.format(foo, bar)) # prints "My favourite food is spam and eggs"
```